src/ape/utils/misc.py

```diff
@@ -490,9 +490,33 @@ def is_zero_hex(address: str) -> bool:
         return False
 
 
+def dict_defaults(mapping: Dict[str, Any], defaults: Dict[str, Any], depth=0):
+    """Conditionally overlay given default on a dict if no value is set"""
+    for key, value in defaults.items():
+        if key not in mapping:
+            mapping[key] = value
+        elif isinstance(value, dict):
+            dict_defaults(mapping[key], value, depth + 1)
+    return mapping
+
+
+def dict_overlay(mapping: Dict[str, Any], overlay: Dict[str, Any], depth=0):
```

Comment:
**thought**: Could make this an internal util method like `_dict_overlay` so we don't have to officially support it from a library perspective and can use it internally as needed. Otherwise, we need a full doc-str here and stuff.

Reply:
I assumed this module was internal use.  Will rename.  Is it fine in this module or is there another spot for 'em?

Reply:
this module is good